File: backend/models/WorkoutLog.cs
using System.ComponentModel.DataAnnotations;

namespace FitTrackBackend.Models;

public class WorkoutLog
{
    [Key]
    public int LogId { get; set; }

    public int WorkoutId { get; set; }

    public int ExerciseId { get; set; }

    public int Sets { get; set; }

    public int Reps { get; set; }

    public decimal Weight { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: FitTrackApp/app/src/main/java/com/example/fittrackapp/models/WorkoutLog.java
package com.example.fittrackapp.models;

public class WorkoutLog {
    public int logId;
    public int workoutId;
    public int exerciseId;
    public int sets;
    public int reps;
    public double weight;

    public WorkoutLog(int workoutId, int exerciseId, int sets, int reps, double weight) {
        this.workoutId = workoutId;
        this.exerciseId = exerciseId;
        this.sets = sets;
        this.reps = reps;
        this.weight = weight;
    }
}